01_financial-well-being/housing-value.qmd

```diff
@@ -0,0 +1,904 @@
+  )
+
+acs1yr |>
+  assertr::verify(valueh < 9999998 | is.na(valueh))
+```
+
+Next, we exclude individuals living in group quarters because we only want households for this metric. In particular, we exclude people living in places other than housing units (e.g., correctional facilities, nursing homes, mental hospitals, dorms, barracks, group homes, missions). For more information, see the [variable description](https://usa.ipums.org/usa-action/variables/GQ#codes_section) on IPUMS USA.
+
+```{r}
+#| label: keep-households-only
+acs1yr <-
+  acs1yr |>
+  filter(!(gq %in% c("Group quarters--Institutions", "Other group quarters")))
+```
+
+FIPS codes should always contain leading zeros so that state codes are two digits, county codes are three digits, and place codes are five digits.
+
+```{r}
+#| label: clean-fips-codes
+
+acs1yr <-
+  acs1yr |>
+  mutate(
+    across(c(state, county, city, puma), as.character),
+    state = str_pad(state, width = 2, pad = "0", side = "left"),
+    county = str_pad(county, width = 3, pad = "0", side = "left"),
+    city = str_pad(city, width = 5, pad = "0", side = "left"),
+    puma = str_pad(puma, width = 5, pad = "0", side = "left")
+  )
+```
+
+We create two age subgroups: households over 45 and households under 45
+
+```{r}
+#| label: add-age-subgroup
+
+acs1yr <-
+  acs1yr |>
+  mutate(subgroup = if_else(age >= 45, "Over or exactly 45 years old", "Under 45 years old"))
+```
+
+We collapse IPUMS race and ethnicity categories into white non-hispanic, Black non-hispanic, other non-hispanic, and
+
+```{r}
+#| label: add-race-ethnicity-subgroup
+
+
+acs1yr <-
+  acs1yr |>
+  mutate(
+    race = lbl_relabel(race, lbl(10, "Other") ~ .lbl %in% c("American Indian or Alaska Native", "Chinese", "Japanese", "Other Asian or Pacific Islander", "Other race, nec", "Two major races", "Three or more major races")),
+    across(c(race, hispan), as_factor)
+  ) |>
+  mutate(
+    race_ethnicity = case_when(
+      race == "White" & hispan == "Not Hispanic" ~ "white_nh",
+      race == "Black/African American" & hispan == "Not Hispanic" ~ "black_nh",
+      race == "Other" & hispan == "Not Hispanic" ~ "other_nh",
+      hispan != "Not Hispanic" ~ "hispanic",
+      .default = NA_character_
+    ),
+    race_ethnicity = factor(race_ethnicity,
+      levels = c(
+        "white_nh",
+        "black_nh",
+        "hispanic",
+        "other_nh"
+      )
+    )
+  )
+```
+
+## Merge Crosswalk {#sec-merge-crosswalk}
+
+We merge our 1-year ACS data with PUMA-county and PUMA-place geocodes to get county and place boundaries.
+
+For 2022 ACS and onward we use a crosswalk going from 2020 PUMA to 2020 county. For pre-2022 ACS data, we use a crosswalk from 2012 PUMA to 2014 county.
+
+The weight of each household is adjusted by the area of the PUMA that falls into a given County/Place
+
+```{r}
+#| label: define-crosswalk-years
+
+acs1yr <-
+  acs1yr |>
+  mutate(crosswalk_period = if_else(year < 2022, "pre-2022", "2022"))
+```
```

Comment:
This may be the start of where you get counting issues later on-- the CT county change that impacts the 2022 data: 
![image](https://github.com/UI-Research/mobility-from-poverty/assets/87652761/5c72eb8e-ac34-4b34-b0c8-dee3d16080d3)

Reply:
I confirmed and added a test that only new FIPS are present in 2022. However, only 6 out of the 9 were present. So I think you're right that this is where the issues with NA's happen down the line.